app/src/vuex/modules/session.js

```diff
@@ -135,21 +155,53 @@ export default ({ apollo }) => {
       }
     },
     async signIn(
-      { state, getters: { currentNetwork }, commit, dispatch },
-      { address, sessionType = `ledger`, networkId }
+      {
+        state,
+        rootState: {
+          connection: { networks },
+        },
+        commit,
+        dispatch,
+      },
+      { address, sessionType = `ledger`, HDPath, curve, networkId }
     ) {
-      if (networkId && currentNetwork.id !== networkId) {
+      const currentNetwork = networks.find(({ id }) => id === networkId)
+      // first search in localStorage for the curve and derivation path
+      const session = JSON.parse(
+        localStorage.getItem(`cosmos-wallets-${address}`)
+      )
+      const finalCrypto = handleSessionCrypto({
+        HDPath,
+        curve,
+        address,
+        session,
+        currentNetwork,
+      })
+      // override crypto with right HDPath and curve
+      HDPath = finalCrypto.HDPath
+      curve = finalCrypto.curve
+      if (networkId && currentNetwork) {
         await commit(`setNetworkId`, networkId)
```

Comment:
is await needed?

Reply:
not sure. `setUserAddresses` had it already. But don't really know